### DISCUSSION TITLE
Store json files in settings sync

### DISCUSSION THREAD

**Discussion by rangav** [microsoft/vscode-discussions#324] [Extension Development QnA] [answered]
Hi All,

I am creator of [Thunder Client](https://marketplace.visualstudio.com/items?itemName=rangav.vscode-thunder-client) extension

Is it possible to `store json files` in `settings sync`, I could not find any docs related to this

it seems there is a size limit for settings sync
https://github.com/microsoft/vscode/issues/86178

**Comment by alefragnani** [comment, marked as answer]
Hi @rangav ,

If you mean _external files_ no, it is not supported. There was an issue requesting something like this, but it was closed recently (https://github.com/microsoft/vscode/issues/113774#issuecomment-1346779712). 

There is however an open issue to sync Workspace State (https://github.com/microsoft/vscode/issues/113996). Maybe this could fit your needs.

Hope this helps

**Reply by rangav** [reply]
Thanks @alefragnani  for the quick reply, so it’s not possible now.

**Reply by alefragnani** [reply]
An alternative would be to store the files in GitHub Gist, and use the Authentication API (https://code.visualstudio.com/api/references/vscode-api#authentication / https://github.com/microsoft/vscode-extension-samples/tree/main/github-authentication-sample) for easier integration. 

That's what I'm planning to do on my Project Manager extension (https://github.com/alefragnani/vscode-project-manager/issues/525).

**Reply by rangav** [reply]
GitHub gist are public only files right? We are trying save api request files, might not suit if it’s public

**Reply by alefragnani** [reply]
No, there are **public** and **private** Gists.

However, I'm not sure if you can define/choose the privacy level of the Gist you store. Maybe something to be defined via Scopes (https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps).

**Reply by rangav** [reply]
We already have feature to save json files in current git repository, one user asked for sync using vscode settings-sync feature without any git related stuff. So gist might not be right use case.

Anyway thanks for the help.
